open-metadata-implementation/access-services/data-engine/data-engine-client/src/main/java/org/odpi/openmetadata/accessservices/dataengine/client/DataEngineEventClient.java

```diff
@@ -0,0 +1,165 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.accessservices.dataengine.client;
+
+import org.odpi.openmetadata.accessservices.dataengine.connectors.intopic.DataEngineInTopicClientConnector;
+import org.odpi.openmetadata.accessservices.dataengine.event.*;
+import org.odpi.openmetadata.accessservices.dataengine.model.*;
+import org.odpi.openmetadata.accessservices.dataengine.model.Process;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.ConnectorCheckedException;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException;
+
+import java.util.List;
+
+
+/***
+ * DataEngineEventClient implements Data Engine client side events interface using provided topic connector.
+ * For more information see {@link DataEngineClient} interface definition.
+ */
+public class DataEngineEventClient implements DataEngineClient {
+
+    private DataEngineInTopicClientConnector topicConnector;
+    private String externalSource;
+
+    /**
+     * Constructor to create DataEngineEventClient with unauthenticated access to the server
+     *
+     * @param source External source system using the client to produce events
+     * @param dataEngineInTopicClientConnector topic connector used to publish to InTopic
+     * @throws InvalidParameterException there is a problem creating the client-side components to issue any
+     *                                   REST API calls.
+     */
+    public DataEngineEventClient(String source, DataEngineInTopicClientConnector dataEngineInTopicClientConnector) {
+        this.topicConnector = dataEngineInTopicClientConnector;
+        this.externalSource = source;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createOrUpdateProcess(String userId, Process process) throws InvalidParameterException, ConnectorCheckedException {
```

Comment:
Maybe it's better if we remove the method definition from the interface & the rest implementation - it is not being used, and we have the corresponding method for creating a list of processes

Reply:
Removed form the interface and rest implementation.